proposals/NightlyBuilds.md

```diff
@@ -0,0 +1,49 @@
+
+Having development version of cordova published on npm would allow everyone to easily set up development version of cordova to test unreleased features, reproduce the bugs or confirm they get fixed in current version.
+
+Nightly builds in our understanding are an npm packages created from current 'master' branches of respective repos and available on NPM under `nightly` dist tag, so they easily could be installed like `npm i cordova@nightly`. All appropriate dependencies of nightly packages also will be updated to point to their nightly versions, e.g. `cordova-cli@6.2.0-nightly` will depend on `cordova-lib@6.2.0-nightly`, etc.
+
+The nightly version of cordova-lib will also has main platform pinned to their nightly versions, so `cordova install android` will automatically install `cordova-android@nightly` to allow early adopters test new features and bugfixes in platforms.
+
+The full format of nightly version is `<package_id>@<next_version>-nightly.<build_date>+<commit_SHA>`, for example
+
+```
+cordova-lib@6.2.0-nightly.2016.05.12+1649800352c51e832262b8469cb4c7f362de28ec
+```
+
+This format is fully Semver-compatible and allows to us to easily recreate the same the cordova setup locally
+
+The nightly packages, we're planning to release, include cordova-cli, cordova-lib and the major platforms (Android, iOS, Windows)
+
+## How it will work
+
+All the aspects of creating and publishing nightly builds are automated using `cordova coho` command. Running `coho nightly` assumes the following steps:
+
+1. For every package to publish update respective repo to the latest commit
+2. For every package update version to nightly version
+  - Also update dependencies versions to point to nightly versions of other `cordova-*` dependencies
+3. For every package run the tests to ensure that release will not contain any known bugs
+4. Publish packages on NPM under `nightly` dist tag
+5. Unpublish old nightly versions of newly published packages
+
+The `cordova coho` command also accepts `--pretend` options to run these steps locally without publishing packages to NPM
+
+## Current status
+
+The work is currently in progress ant the following items are completed:
+
+1. The [Jenkins job](https://builds.apache.org/view/A-D/view/Cordova/job/cordova-nightly/) that has npm set up using credentials for publishing packages
+2. The [update](https://github.com/apache/cordova-coho/pull/116) for `coho nightly` that supports publishing of 'cordova-cli' and `cordova-lib`
+3. Patches for `cordova-cli` and `cordova-lib` that fixes some tests that were failing when were running on nightly builds (_patches are on their go and will be submitted soon_)
+
+## Next steps
+
+1. Update cordova to print necessary warnings and information about nightly versions of dependencies and platforms when running `cordova --version` and `cordova platform ls`
+2. Add support to `coho` for building nightly versions of platforms

```

Comment:
Perhaps we need to also mention on our website that a nightly build can be used.